packages/provider/src/RelayProvider.ts

```diff
@@ -131,7 +141,33 @@ export class RelayProvider implements HttpProvider, Web3ProviderBaseInterface {
     })
   }
 
-  _ethSendTransaction (payload: JsonRpcPayload, callback: JsonRpcCallback): void {
+  /**
+   * The ID can be either a RelayRequestID which requires event-based lookup or Transaction Hash that goes through
+   * @param payload
+   * @param callback
+   */
+  async _ethGetTransactionReceipt (payload: JsonRpcPayload, callback: JsonRpcCallback): Promise<void> {
+    const id = (typeof payload.id === 'string' ? parseInt(payload.id) : payload.id) ?? -1
+    const relayRequestID = payload.params[0] as string
+    const submissionDetails = this.submittedRelayRequests.get(relayRequestID)
+    if (submissionDetails == null) {
+      this._ethGetTransactionReceiptWithTransactionHash(payload, callback)
+      return
+    }
+    try {
+      const result = await this._createTransactionReceiptForRelayRequestID(relayRequestID, submissionDetails)
+      const rpcResponse = {
+        id,
```

Comment:
why do you translate id to int ? I think its better to return id as it is (and also jsonrpc) from the original request.

Reply:
Something broke down when I didn't. I can try again. I didn't do it out of pure passion for integers if I recall correctly.